Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp

```diff
@@ -342,11 +381,18 @@ namespace Terrain
             point.m_quadMeshType = terrainExists ? Physics::QuadMeshType::SubdivideUpperLeftToBottomRight : Physics::QuadMeshType::Hole;
             Physics::MaterialId materialId = FindMaterialIdForSurfaceTag(surfaceWeight.m_surfaceType);
             point.m_materialIndex = GetMaterialIdIndex(materialId, materialList);
-            heightMaterials.emplace_back(point);
+
+            size_t lookUpIndexInVector = xOffset + xIndex + (yOffset + yIndex) * gridWidth;
+
+            AZ_Assert(lookUpIndexInVector < heightMaterials.size(), "Heightmaterial index is out of range!");
+            if (lookUpIndexInVector < heightMaterials.size())
```

Comment:
If it's an assert then it's a programmer error, it should never happen and the execution status is undetermined from there, no need to do extra check with `if` then.

On the other hand, if it's a potential error that can happen and be recovered from, then keep the `if` and replace az_assert with az_error.